x/emissions/migrations/v3/migrate.go

```diff
@@ -86,11 +93,184 @@ func MigrateParams(store storetypes.KVStore, cdc codec.BinaryCodec) error {
 		HalfMaxProcessStakeRemovalsEndBlock: oldParams.HalfMaxProcessStakeRemovalsEndBlock,
 		EpsilonSafeDiv:                      oldParams.EpsilonSafeDiv,
 		DataSendingFee:                      oldParams.DataSendingFee,
-		// NEW PARAMS
-		MaxElementsPerForecast: defaultParams.MaxElementsPerForecast,
+		MaxElementsPerForecast:              defaultParams.MaxElementsPerForecast,
+		MaxActiveTopicsPerBlock:             defaultParams.MaxActiveTopicsPerBlock,
 	}
 
 	store.Delete(types.ParamsKey)
 	store.Set(types.ParamsKey, cdc.MustMarshal(&newParams))
 	return nil
 }
+
+func MigrateActiveTopics(store storetypes.KVStore, ctx sdk.Context, emissionsKeeper keeper.Keeper) error {
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	sdkCtx.Logger().Warn("MigrateActiveTopics")
+	topicStore := prefix.NewStore(store, types.TopicsKey)
+	topicFeeRevStore := prefix.NewStore(store, types.TopicFeeRevenueKey)
+	topicStakeStore := prefix.NewStore(store, types.TopicStakeKey)
+	topicPreviousWeightStore := prefix.NewStore(store, types.PreviousTopicWeightKey)
+	iterator := topicStore.Iterator(nil, nil)
+	churningBlockStore := prefix.NewStore(store, types.TopicToNextPossibleChurningBlockKey)
+	blockToActiveStore := prefix.NewStore(store, types.BlockToActiveTopicsKey)
+	blockLowestWeightStore := prefix.NewStore(store, types.BlockToLowestActiveTopicWeightKey)
+	params, err := emissionsKeeper.GetParams(ctx)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get params for active topic migration")
+	}
+	churningBlock := make(map[types.TopicId]types.BlockHeight, 0)
+	blockToActiveTopics := make(map[types.BlockHeight]types.TopicIds, 0)
+	lowestWeight := make(map[types.BlockHeight]types.TopicIdWeightPair, 0)
+
+	topicWeightData := make(map[types.TopicId]alloraMath.Dec, 0)
+
+	for ; iterator.Valid(); iterator.Next() {
+		var oldMsg oldtypes.Topic
+		err := proto.Unmarshal(iterator.Value(), &oldMsg)
+		if err != nil {
+			continue
+		}
+		var feeRevenue = cosmosMath.NewInt(0)
+		idArray := make([]byte, 8)
+		binary.BigEndian.PutUint64(idArray, oldMsg.Id)
+		err = feeRevenue.Unmarshal(topicFeeRevStore.Get(idArray))
+		if err != nil {
+			continue
+		}
+		var stake = cosmosMath.NewInt(0)
+		err = stake.Unmarshal(topicStakeStore.Get(idArray))
+		if err != nil {
+			continue
+		}
+		var previousWeight = alloraMath.NewDecFromInt64(0)
+		err = previousWeight.Unmarshal(topicPreviousWeightStore.Get(idArray))
+		if err != nil {
+			continue
+		}
+		// Get topic's latest weight
```

Comment:
Nbd, but you can improve the code above: instead of doing `NewStore` then `Unmarshal`, you can just use the emissionsKeeper, e.g., `GetPreviousTopicWeight`. Same thing for setting, e.g: `SetPreviousTopicWeight`